### PR TITLE
Update UG

### LINKED ISSUE #232: [PE-D][Tester B] Filter does not align with UG’s behaviour
Screenshot 2024-11-08 at 3.48.10 PM

Steps to Reproduce:
`filter p/99`

Expected: Based on UG filter `p/99` should return `99999999`
Actual: Error message for phone number constraints
Suggestions: Update in UG that phone number should be at least 3 digits long

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: Sornsornah/ped#2

### LINKED ISSUE #242: [PE-D][Tester F] Application screenshot on user guide for filter command does not match actual application
The image showcasing the results of using the filter command (on page 6) differs from the display that is in the application itself, as shown in the screenshot below. Suggest updating the image in the User Guide to the updated version of the application.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: kevin-pek/ped#1

### LINKED ISSUE #251: [PE-D][Tester C] Filter description is too long and unintuitive
There's too much text here for me to understand what is really going on. I suggest to refine it down. There's also an inconsistency in using `code` brackets for `AND` search and 'OR' search. I'm still confused on why multiple predicated and multiple values results in both an OR and AND search.

Furthermore, the use of the word Predicate might be too technical for non-technical users.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: anselmlong/ped#12

### LINKED ISSUE #256: [PE-D][Tester A] `addEcName` inputs lack support for legal names with `/`
Certain characters, such as /, are currently disallowed in the name input fields where they may interfere with command parsing. It may disrupt user experience, especially when the restricted characters are part of a legally required name format.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: SeeYangZhi/ped#6

### LINKED ISSUE #260: [PE-D][Tester B] UG delete command has invalid example
Issue: Example includes `find Betsy`, but there is no `find` command
Suggestion: Change to `filter` instead

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: Sornsornah/ped#9

### LINKED ISSUE #272: [PE-D][Tester D] Strange blue line on the top of Quick Start gui image
In the quick start section of your UG with a sample image of your gui, there is a blue artifact (?) line(?) at the top of the image which I have circled above. Nothing major but it would be good to correct it as it may be interpreted as part of the UI.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: flyingsalsa/ped#6

### LINKED ISSUE #277: [PE-D][Tester E] Filtering by Address
# Steps to Reproduce
1. Use the default contact list
2. Do filter a/blk 30

# Expected Outcome
Displays Alex Yeoh and Bernice Yu only as their address has Blk 30.
# Actual Outcome
The other contacts are also displayed.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: george-yeo/ped#5

### LINKED ISSUE #281: [PE-D][Tester E] Adding Indian names
# Steps to Reproduce
1. Add a contact with Name containing "s/o", used commonly in Indian names, like add n/Raja s/o Ramesh p/22224444 e/raja@example.com a/123, Clementi Rd, 1234665 c/1A s/M r/1

# Expected Outcome
Contact successfully added.

# Actual Outcome
Error message regarding s/

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: george-yeo/ped#6

### LINKED ISSUE #283: [PE-D][Tester C] Wrong capitalisation
> : How do I transfer my data to another Computer?

computer*

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: anselmlong/ped#27

### LINKED ISSUE #295: [PE-D][Tester F] Filter command usage for phone number does not match user guide
In the user guide (page 4), the following example was given to show how the filter command can be used to filter students by their phone number:

However, trying to follow this example leads to the following error message being shown on the application:

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: kevin-pek/ped#4

### LINKED ISSUE #321: [PE-D][Tester E] Filtering not working as expected with Phone
# Steps to Reproduce
1. Filter using filter p/99

# Expected Outcome
Contacts containing "99" in their Phone number should be shown, as described by the user guide.

# Actual Outcome
Error message is thrown

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: george-yeo/ped#12

### LINKED ISSUE #322: [PE-D][Tester D] Emergency contact names can have no letters at all
Inputting the command `addEcName 1 en/12123` allows for Emergency contact name with just numbers which is strange since then i would be allowed to enter the the emergency phone number as both the name and the phone number

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: flyingsalsa/ped#11

### LINKED ISSUE #325: [PE-D][Tester E] Filtering example is not updated to the app
The ui in the screenshot is not updated to the current ui version.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: george-yeo/ped#13

### LINKED ISSUE #334: [PE-D][Tester D] Only complete word matches allowed for filtering with names
I know it has been mentioned already that only full word matches are allowed but doesn't this hamper the functionality? What if you cannot remember the exact spelling of a person's name?

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: flyingsalsa/ped#15

### LINKED ISSUE #336: [PE-D][Tester D] 99 is an invalid phone number
example in the UG pg 4 under the filter section is incorrect we can cannot filter using p/99 in the first place

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: flyingsalsa/ped#17

### LINKED ISSUE #342: [PE-D][Tester D] Allowing blank fields but inability to search for them
What if I want to filter all students that haven't filled in their emergency contacts yet so I can notify them to do so? being unable to filter this condition is troublesome to the user.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: flyingsalsa/ped#22

### PR DESCRIPTION
**Pull request by Shruthiiii03**
Updated UG using feedback from issues:

1) Refining emergency contact attribute description in filter 
Fixes #342 

2) Correct phone number filter description
Fixes #336 
Fixes #321 
Fixes #295
Fixes #232

3) Add parameter restrictions for EcNames and Names: no legal operators, numbers allowed
Fixes #322 
Fixes #281 
Fixes #256 

4) Correct Address filtering requirements 
Fixes #277 

5) Improve filter section readability 
Fixes #251

6) Others
Fixes #334 
Fixes #325   
Fixes #283 
Fixes #272
Fixes #260
Fixes #242

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ StudentManagerPro (SMP) is a **desktop app for managing students, optimized for
 
    * `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01 c/1A s/M r/1` : Adds a student named `John Doe` to the student list.
 
+   * `delete 3` : Deletes the 3rd student contact shown in the current list.
```
Comment (gohqingkhang) [inline docs/UserGuide.md:35]:
```suggestion

```
is this a duplicate?

**Review by gohqingkhang — COMMENTED** [review]
LGTM, small issue

File: docs/UserGuide.md (hunk 1 above)
Comment (Shruthiiii03) [reply docs/UserGuide.md:35]: Yup let me remove it thanks!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -176,7 +186,7 @@ Format: `delete INDEX`
 
 Examples:
 * `list` followed by `delete 2` deletes the 2nd person in the student list.
-* `find Betsy` followed by `delete 1` deletes the 1st person in the results of the `find` command.
+* `filter Betsy` followed by `delete 1` deletes the 1st person in the results of the `filter` command.
```
Comment (BunnyHoppp) [inline docs/UserGuide.md:189]: Should be `filter n/Betsy`

**Pull request merged**